Oneirocom/Magick
# Remote plugin

## What Changed:

Large graphs were problematic on the client because on every save, the whole graph would get reloaded into memory.  This was because we had the nodes subscription occurring inside the worker, which would be rerun on save because of the task plugin.  

Since we run all the work on the server, there was no need to have the worker hold the subscription to the node's work on the server.  It also meant I could remove the task plugin completely from the client.

To do this, I wrote a new RemotePlugin which subscribes to individual node events in the builder so it only hooks in once.  I got rid of the consolePlugin, SocketOverridePlugin and the EmitPlugin.  I also switched from using the raw socket connection to using an event on the agent service for sending spell-running information.  

I installed the remote plugin on the shared engine as well only on the server.  It uses an emit function passed in which lives on the spellRunner.  This currently behaves differently if loaded on the server vs on an agent. However, once we move this off the server onto an agent it should still function well.

Now there should be no lag when working with extremely large graphs on the client side.

## How to test:

Use the client to run spells from the playtest. Everything should operate as expected with no changes.

## Additional information:

Any other information that might be useful while reviewing.
